lib/controllers/project.js

```diff
@@ -21,779 +22,720 @@
+Project.prototype.clearLogs = function (projectId, callback) {
+  Librarian.project.clearLogs(
+    projectId, UserConfig.data.apiKey, function (err, res) {
+      if (err) return callback(Errors.getMessage(err));
+      if (res.statusCode === NOT_FOUND) {
+        return callback('Problem clearing logs: ' + res.statusCode);
+      }
+      callback(null, res);
+    });
 };
 
-/**
+/*
  * Streams log for the specified project.
  * @param {String} projectId The project ID.
  * @param {function} callback.
  */
-//-----------------------------------------------------------------------------
-Project.prototype.streamLogs = function(projectId, servos, callback){
+// ----------------------------------------------------------------------------
+Project.prototype.streamLogs = function (projectId, servos, callback) {
   // TODO: add param for prefix?
-  modulus.io.success('Log streaming started (ctrl-c to exit)...');
-  if ( ! Array.isArray(servos)) {
-    servos = [servos];
-  }
-
-  servos.forEach(function (servo){
-    var endpoint = util.format('%s://%s:%s/project/%s/logs/stream?authToken=%s&servoId=%s',
-      librarian._http._ssl ? 'https' : 'http',
-      librarian._http._host,
-      librarian._http._port,
+  Modulus.io.success('Log streaming started (ctrl-c to exit)...');
+  if (!Array.isArray(servos)) servos = [servos];
+
+  servos.forEach(function (servo) {
+    var endpoint, h, done, tr, req, disconnected = false;
+    endpoint = Util.format('%s://%s:%s/project/%s/logs/stream?authToken=%s&servoId=%s',
+      Librarian._http._ssl ? 'https' : 'http',
+      Librarian._http._host,
+      Librarian._http._port,
       projectId,
-      userConfig.data.apiKey,
+      UserConfig.data.apiKey,
       servo.id);
 
-    var h = librarian._http._ssl ? https : http;
-    var disconnected = false;
+    h = Librarian._http._ssl ? Https : Http;
+    disconnected = false;
 
-    var done = function () {
-      if ( ! disconnected) {
-        callback();
+    done = function () {
+      if (!disconnected) {
         disconnected = true;
+        return callback();
       }
     };
 
-    var tr = through(function (line) {
-      var prefix = servo.id.substr(0, 8);
-      line = util.format('%s> %s\n', prefix, line);
+    tr = Through(function (line) {
+      var prefix = servo.id.substr(0, PREFIX_SUBSTRING);
+      line = Util.format('%s> %s\n', prefix, line);
       this.queue(line);
     });
 
-    var req = h.get(endpoint, function (res) {
-      if (servos.length > 1) {
-        res.pipe(split()).pipe(tr).pipe(process.stdout);
-      } else {
-        res.pipe(process.stdout);
-      }
-      res.on('error', function () { done(); });
-      res.on('close', function () { done(); });
-      res.on('end', function () { done(); });
+    req = h.get(endpoint, function (res) {
+      if (servos.length > 1) res.pipe(Split()).pipe(tr).pipe(process.stdout);
+      else res.pipe(process.stdout);
+      res.on('error', function () {
+        done();
```

Comment:
`done` doesn't care about arguments, so all of these could just be: `res.on(EVENT, done)`

...maybe for refactor branch